libfreerdp/core/peer.c

```diff
@@ -1247,11 +1248,12 @@ static BOOL freerdp_peer_send_server_redirection_pdu(freerdp_peer* peer,
 		goto fail;
 	if (!rdp_send_pdu(peer->context->rdp, s, PDU_TYPE_SERVER_REDIRECTION, 0))
 		goto fail;
```

Comment:
Maybe use `goto out`. Make `rc = rdp_reset_runtime_settings(peer->context->rdp);` . And `return rc`